### PR TITLE
TRITON-2242 zfs.js needs to be reconciled across smartos-live, node-zfs and node-zfs/linuxcn

### PR DESCRIPTION
**Pull request by bahamat**
This includes changes from smartos-live for the following:
    
* OS-1751
* OS-7698
* OS-8198

### CODE REVIEW

File: lib/zfs.js [hunk 1]
```diff
@@ -140,13 +151,41 @@ zpool.status = function (pool, callback) {
...
+	var cmd;
 	var args;
 
-	if (arguments.length != 3)
-		throw Error('Invalid arguments, 3 arguments required');
+	if (arguments.length === 3) {
+		callback = force;
+		encrypt = false;
+		force = false;
+		efi = true;
```
Comment (danmcd) [inline lib/zfs.js:163]: For the linux-cn branch, we may need to set this to false as I'm not sure zpool has a -B on Linux.  OTOH if we only ever access this through mkzpool, we explicitly set "efi" from the caller, and can therefore DTRT.

**Review by danmcd — DISMISSED** [review]
Seems okay modulo one warning, but if we always use mkzpool it's far less concerning.

File: lib/zfs.js (hunk 1 above)
Comment (bahamat) [reply lib/zfs.js:163]: I'll do an `if` clause for Linux.

File: lib/zfs.js [hunk 2]
```diff
@@ -140,13 +152,41 @@ zpool.status = function (pool, callback) {
...
-	args = [ 'create', pool ];
+	if (encrypt === true) {
+		cmd = exports.paths.kbmadm;
+		args = [ 'create-zpool' ];
+	} else {
+		cmd = exports.paths.zpool;
+		args = [ 'create' ];
+	}
+
+	if (efi === true) {
```
Comment (danmcd) [inline lib/zfs.js:182]: Linux check here?!?

Comment (bahamat) [reply lib/zfs.js:182]: Yeah.

Comment (danmcd) [reply lib/zfs.js:182]: Let the record show these changes will be going into the linuxcn branch.

**Review by danmcd — APPROVED** [review]

**Pull request merged**